specification.md

```diff
@@ -1055,10 +1057,168 @@ For example:
 }
 ```
 
-Note that the referenced function definition type in this case must be `rest` (default type).
+Note that the referenced function definition type in this case must be `openapi` (default type).
 
 For more information about functions, reference the [Functions definitions](#Function-Definition) section.
 
+#### Using functions for RESTful Service Invocations
+
+The specification also supports describing REST invocations in the [functions definition](#Function-Definition) using [OpenAPI Paths Object](https://spec.openapis.org/oas/v3.1.0#paths-object).
+
+Here is an example function definition for REST requests with method `GET` and request target corresponding with [URI Template](https://www.rfc-editor.org/rfc/rfc6570.html) `/users/{id}`:
+
+```json
+{
+  "functions":[
+    {
+      "name":"queryUserById",
+      "operation": {
+        "/users": {
+          "get": {
+            "parameters": [{
+              "name": "id",
+              "in": "path",
+              "required": true 
+            }]
+          }
+        }
+      },
+      "type":"rest"
+    }
+  ]
+}
+```
+
+Note that the [Function Definition](#Function-Definition)'s `operation` property must follow the OpenAPI Paths Object specification definition.
+
+The function can be referenced during workflow execution when the invocation of the REST service is desired. For example:
+
+```json
+{
+  "states":[
+    {
+      "name":"QueryUserInfo",
+      "type":"operation",
+      "actions":[
+        {
+          "functionRef":"queryUserById",
+          "arguments":{
+            "id":"${ .user.id }"
+          }
+        }
+      ],
+      "end":true
+    }
+  ]
+}
+```
+
+Example of the `POST` request sending the state data as part of the body:
+
+```json
+{
+  "functions":[
```

Comment:
this is 30 lines of code and looks super ugly inside a workflow definition itself...and this is a small example. i really think we should not show this to users and use spring operation to a file that contains the object needed

Reply:
locala file is ok ..or even expression. i honestly do not like this when looking at it at all :(

Reply:
Yeah.. it's pretty big, but it's a possibility for a more complex operation, and I believe we should keep it to illustrating. Since we do support:

```json
{
    "functions": "file://myfuncs.json"
}
```

It is possible to externalize these definitions. Especially now with #691. wdyt?

Reply:
> this is 30 lines of code and looks super ugly inside a workflow definition itself...and this is a small example. i really think we should not show this to users and use spring operation to a file that contains the object needed

@tsurdilo I totally agree with you, man! However, let's not forget this function type (and maybe this should be emphazied somewhere) is but for compatibility with "legacy" APIs, or undocumented ones over which the consumer has no power. In other words, one should (and probably would) always prefer using the `openapi` function type.

> It is possible to externalize these definitions. Especially now with #691. wdyt?

It is possible, indeed, but I would STRONGLY advise against it, as doing so would be equivalent to using a proper OpenAPI documentation file, thus using the `openapi` function type instead.

Reply:
I agree. We can either add a note there adding this point or not mention it at all. I don't think we mention it.